classes/class-gf-securesubmit.php

```diff
@@ -1705,144 +1705,141 @@ public function subscribe($feed, $submission_data, $form, $entry)
+            if (null === $payPlanPaymentMethod->paymentMethodKey) {
 
-                    } else {
+                $this->log_debug(__METHOD__ . '(): Could not create Pay Plan payment method');
 
-                        // Get HPS plan for feed.
-                        $this->log_debug(__METHOD__ . '(): Create Schedule.');
-                        /** @var HpsPayPlanSchedule $plan */
-                        $plan = $this->create_plan($payPlanPaymentMethod, $feed, $payment_amount, $trial_period_days, $currency);;
+                return $this->authorization_error($userError);
 
-                        // If error was returned when retrieving plan, return plan.
+            }
 
-                        /** @var HpsPayPlanSchedule $planSchedule */
-                        $this->log_debug(__METHOD__ . '(): Add Schedule.');
-                        $planSchedule = $payPlanService->addSchedule($plan);
+            // Get HPS plan for feed.
+            $this->log_debug(__METHOD__ . '(): Create Schedule.');
+            /** @var HpsPayPlanSchedule $plan */
+            $plan = $this->create_plan($payPlanPaymentMethod, $feed, $payment_amount, $trial_period_days,
+                $currency);;
 
-                        if (!isset($subscribResult) || null === rgar($subscribResult, 'error_message')) {
+            // If error was returned when retrieving plan, return plan.
 
-                            // Create the plan unless there is no key.
-                            if (null === $planSchedule->scheduleKey) {
+            /** @var HpsPayPlanSchedule $planSchedule */
+            $this->log_debug(__METHOD__ . '(): Add Schedule.');
+            $planSchedule = $payPlanService->addSchedule($plan);
 
-                                $this->log_debug(__METHOD__ . '(): Could not create Pay Plan Schedule');
-                                $subscribResult = $this->authorization_error($userError);
+            if (!isset($subscribResult) || null === rgar($subscribResult, 'error_message')) {
 
-                            } else {
+                // Create the plan unless there is no key.
+                if (null === $planSchedule->scheduleKey) {
 
-                                if (null !== $trial_period_days) {
+                    $this->log_debug(__METHOD__ . '(): Could not create Pay Plan Schedule');
 
-                                    $this->log_debug(__METHOD__ . '(): Processing one time setup fee');
-                                    /** @var HpsAuthorization $response */
-                                    /** @noinspection PhpParamsInspection */
-                                    $response = $this->processRecurring($payment_amount,$feed,$payPlanPaymentMethod,$planSchedule);
+                    return $this->authorization_error($userError);
 
-                                    if (!($response->transactionId > 0 && null !== $response->authorizationCode)) {
+                } else {
```

Comment:
Since the `if` portion that precedes this `else block returns early, this isn't required.